common/djangoapps/third_party_auth/models.py

```diff
@@ -810,6 +810,18 @@ def match_social_auth(self, social_auth):
         prefix = self.slug + ":"
         return self.backend_name == social_auth.provider and social_auth.uid.startswith(prefix)
 
+    def get_remote_id_from_field_name(self, social_auth, field_name):
+        """ Given a UserSocialAuth object, return the user remote ID against the field name provided. """
+        try:
+            assert self.match_social_auth(social_auth)
```

Comment:
I think the assert statement is unnecessary here, and except for returning None, we should raise the exception, as on L827 will raise an exception. 

How about handling it this way?

```
def get_remote_id_from_field_name(self, social_auth, field_name):
    """Given a UserSocialAuth object, return the user remote ID against the field name provided."""
    if not self.match_social_auth(social_auth):
        raise ValueError(
            f"Invalid social_auth object provided: {social_auth}. "
            "It does not match the expected provider."
        )
    field_value = social_auth.extra_data.get(field_name)
    if field_value and isinstance(field_value, list):
        return field_value[0]
    return field_value
```